pulp_2to3_migration/app/migration.py

```diff
@@ -234,6 +241,22 @@ def create_repoversions_publications_distributions(plan, parallel=True):
         parallel (bool): If True, attempt to migrate things in parallel where possible.
     """
     for plugin in plan.get_plugin_plans():
+        # very whether all pulp2 repos and distributors have been migrated
+        not_migrated_repos = Pulp2Repository.objects.filter(
+            is_migrated=False,
+            not_in_plan=False,
+            pulp2_repo_type=plugin.type)
+        not_migrated_distributors = Pulp2Distributor.objects.filter(
+            is_migrated=False,
+            not_in_plan=False,
+            pulp2_type_id__in=plugin.migrator.distributor_migrators.keys())
+        # no need to proceed - everything is migrated
+        if not not_migrated_repos and not not_migrated_distributors:
+            break
```

Comment:
continue?

Reply:
I guess one other minor nitpick, we could just do this query once for all types, and then sort the results by type manually, to avoid multiple queries, since we need all of these results anyway.

Not a huge difference, though, if you excluded it I wouldn't mind.

Reply:
I did this per plugin because i don't expect the plan to contain a lot of plugins migrated at once.
But your point is valid i just did not want to go through manual sorting